src/Strategy/Node/PackageJson.hs

```diff
@@ -85,3 +90,83 @@ buildGraph PackageJson{..} = run . withLabeling toDependency $ do
         , dependencyEnvironments = mempty
         , dependencyTags = Map.empty
         }
+
+newtype PkgJsonWorkspaces = PkgJsonWorkspaces {unWorkspaces :: [Glob Rel]}
+  deriving (Eq, Ord, Show, ToJSON)
+
+-- Name and version are required for workspace sub-projects.
+data PackageJson = PackageJson
+  { packageName :: Maybe Text
+  , packageVersion :: Maybe Text
+  , packageWorkspaces :: PkgJsonWorkspaces
+  , packageDeps :: Map Text Text
+  , packageDevDeps :: Map Text Text
+  }
+  deriving (Eq, Ord, Show)
+
+instance FromJSON PkgJsonWorkspaces where
+  parseJSON (Array x) = PkgJsonWorkspaces <$> parseJSON (Array x)
+  parseJSON (Object x) = withObject "PkgJsonWorkspaces" go (Object x)
+    where
+      -- We might find a "nohoist" key, but it only tells us where to find
+      -- installed deps, rather than which deps are installed.
+      -- https://classic.yarnpkg.com/blog/2018/02/15/nohoist/
+      go obj = PkgJsonWorkspaces <$> obj .:? "packages" .!= []
+  parseJSON _ = fail "'workspaces' must be an array or an object"
+
+instance FromJSON PackageJson where
+  parseJSON = withObject "PackageJson" $ \obj ->
+    PackageJson <$> obj .:? "name"
+      <*> obj .:? "version"
+      <*> obj .:? "workspaces" .!= PkgJsonWorkspaces []
+      <*> obj .:? "dependencies" .!= Map.empty
+      <*> obj .:? "devDependencies" .!= Map.empty
+
+instance ToJSON PackageJson where
+  toJSON PackageJson{..} =
+    object
+      [ "name" .= packageName
+      , "version" .= packageVersion
+      , "workspaces" .= packageWorkspaces
+      , "dependencies" .= packageDeps
+      , "devDependencies" .= packageDevDeps
+      ]
+
+type Manifest = Path Abs File
+type FileGraph = AM.AdjacencyMap Manifest
```

Comment:
These are intentionally type aliases.  I was just going to use the types as-is, but I hated writing `AM.AdjacencyMap (Path Abs File)` everywhere, especially since vscode was being shitty about autocomplete.